splink/term_frequencies.py

```diff
@@ -75,14 +75,58 @@ def _join_tf_to_input_df_sql(linker: Linker):
     return sql
 
 
-def term_frequencies_from_concat_with_tf(input_column):
+def _join_new_table_to_df_concat_with_tf_sql(linker: Linker, new_tablename) -> str:
+    # If the user provides a new table e.g. when using linker.compare_two_records
+    # or linker.find_matches_to_new_records we need to join on term frequencies
+    # The easiest way to do this is to use registered tf tables
+    # if they exist or obtain tf joining to __splink__df_concat_with_tf
+    # if not
+    # If __splink__df_concat_with_tf does not exist, we fall back further
+    # to not using tf adjustments by inserting nulls into the tf columns
+
+    cache = linker._intermediate_table_cache
+    settings_obj = linker._settings_obj
+    tf_cols = settings_obj._term_frequency_columns
+
+    select_cols = [f"{new_tablename}.*"]
+
+    for col in tf_cols:
+        tbl = colname_to_tf_tablename(col)
+        if tbl in cache:
+            select_cols.append(f"{tbl}.{col.tf_name}")
+
+    template = "left join {tbl} on " + new_tablename + ".{col} = {tbl}.{col}"
+    template_with_alias = (
+        "left join ({subquery}) as {_as} on " + new_tablename + ".{col} = {_as}.{col}"
+    )
+
+    left_joins = []
+    for i, col in enumerate(tf_cols):
+        tbl = colname_to_tf_tablename(col)
+        if tbl in cache:
+            sql = template.format(tbl=tbl, col=col.name)
+            left_joins.append(sql)
+        elif "__splink__df_concat_with_tf" in cache:
+            subquery = f"""
+            select distinct {col.name}, {col.tf_name}
```

Comment:
I've moved the logic that derives tf columns from `__splink__df_concat_with_tf` to here

This logic is no longer used in `linker.compute_tf_table` because it's rare to need to 'backwards induce' tf tables in this way, and it adds complexity for little computational gain (tf tables are fast to compute)